### ISSUE TITLE
Feature: mixlib-shellout trace output

### ISSUE THREAD

**Issue by jeremiahsnapp** [chef/chef#3590]
Users want shellout to display the command that was run, STDOUT, STDERR and return status when trace logging is enabled. Basically the output given by `Mixlib::ShellOut#format_for_exception` but this is only displayed when the return status doesn't match the expected return status.

 `Mixlib::ShellOut#format_for_exception` code

https://github.com/chef/mixlib-shellout/blob/master/lib/mixlib/shellout.rb#L225-L234

As a workaround I'm able to put the following code in `libraries/helper.rb` in a cookbook to monkeypatch the functionality so it's available for all shellout usage.

```
module Mixlib
  class ShellOut
    def run_command
      if logger
        log_message = (log_tag.nil? ? "" : "#@log_tag ") << "sh(#@command)"
        logger.send(log_level, log_message)
      end
      result = super
      ::Chef::Log.trace format_for_exception if Chef::Config[:log_level] == :trace
      result
    end
  end
end
```

**NOTE: this issue is very old and references Mixlib::ShellOut#run_command which has been removed**

For a real implementation of this feature at least the following things will need to be done.
- [mixlib-shellout](https://github.com/chef/mixlib-shellout) needs the ability to display the output when its log_level is set to trace.
- shellout usage in chef needs to pass `Chef::Config[:log_level]` as an option to the shellout command because currently it is being statically set to `:info` for the execute resource and `:trace` for things like the service resource

**Comment by lamont-granquist** [comment]
sorta related to #2788 probably.

**Comment by damacus** [comment]
Looks like this might have been fixed by #4040

**Comment by lamont-granquist** [comment]
yeah think this got done
